openmmforcefields/generators/template_generators.py

```diff
@@ -1589,6 +1589,14 @@ def generate_residue_template(self, molecule, residue_atoms=None):
         system = self._smirnoff_forcefield.create_openmm_system(
             molecule.to_topology(), charge_from_molecules=charge_from_molecules
         )
+
+        # Remove CMMotionRemover if present
+        # See https://github.com/openmm/openmmforcefields/issues/365
+        for f_idx in range(system.getNumForces()):
+            force = system.getForce(f_idx)
+            if "CMMotionRemover" in str(force):
+                system.removeForce(f_idx)
```

Comment:
A little safer - hard to expect multiple CMM removers, but this is modifying a list while iterating over it. Not sure what happens if, for example, the force is listed last

```suggestion
                break
        system.removeForce(f_idx)
```

Reply:
Loop in reverse order from `system.getNumForces()-1` down to 0.  That way the indices will still be correct if you remove one.

The best way to check the force type is `if isinstance(force, CMMotionRemover)`.

Why are you removing CMMotionRemovers?

Reply:
The check here flags it as a custom force: https://github.com/openmm/openmmforcefields/blob/7a32d2d109a6cb44dd4344ef0482d7cf7f352cdd/openmmforcefields/generators/template_generators.py#L1116-L1131

Reply:
That means you should fix the check, not remove the force.  The default for `createSystem()` is `removeCMMotion=True`.  This change would break it.

Reply:
This check happens when generating the residue template from OpenFF tooling, not the system that's ultimately passed to the user

Reply:
I would need to look at the code more closely, but what happens if when iterating over `system.getForces()` you just `continue` if the force is `CMMotionRemover`? That should keep it out of the `forces` dict?

Reply:
Okay so if we go with my idea, `CMMotionRemover` won't end up in the `ffxml`  when you call `generate_residue_template` -- is that a problem? If it is, then all we need to do is write `CMMotionRemover` into the `ffxml`

Reply:
Strong preference for 1